Backoffice/Security/Authorization/Voter/AbstractPerimeterVoter.php

```diff
@@ -35,16 +36,33 @@ public function __construct(PerimeterManager $perimeterManager)
      */
     protected function isSubjectInPerimeter($subjectKey, UserInterface $user, $entityType)
     {
-        foreach ($user->getGroups() as $group) {
-            $perimeter = $group->getPerimeter($entityType);
-
-            if ($perimeter instanceof PerimeterInterface
-                && $this->perimeterManager->isInPerimeter($subjectKey, $perimeter)
-            ) {
-                return true;
+        if (false === $this->hasCachedPerimeters($user, $entityType)) {
+            $cachedPerimeter = $this->perimeterManager->createPerimeter($entityType);
+            foreach ($user->getGroups() as $group) {
+                $perimeter = $group->getPerimeter($entityType);
+                if ($perimeter instanceof PerimeterInterface) {
+                    $cachedPerimeter->addItems($perimeter->getItems());
+                }
             }
+
+            $this->cachedPerimeters[$user->getId()][$entityType] = $cachedPerimeter;
         }
 
-        return false;
+        return $this->perimeterManager->isInPerimeter($subjectKey, $this->cachedPerimeters[$user->getId()][$entityType]);
```

Comment:
@alavieille : to increase the reusability of the cache feature, you could isolate lines 39 to 49 in a protected method returning the cached perimeters and named ``getAllPerimeters`` or ``getAglomeratedPerimeter``. That method could be used in line 51. That way, the 'master perimeter' could be reuse later outside isSubjectInPerminter without knowing if the cache exists or must generated.